app/services/katello/pulp3/repository/docker.rb

```diff
@@ -4,7 +4,7 @@ module Katello
   module Pulp3
     class Repository
       class Docker < ::Katello::Pulp3::Repository
-        def api_client
+        def self.api_client(smart_proxy)
           PulpDockerClient::ApiClient.new(smart_proxy.pulp3_configuration(PulpDockerClient::Configuration))
         end
 
```

Comment:
You may want to look at https://github.com/Katello/katello/pull/8224 and add a similar RepositorySyncURL class for docker content. Also, you can test progress presenter with https://github.com/Katello/katello/pull/8223 and see if it works with docker content.